docs/rules/rrd/v-for-with-index-key.md

```diff
@@ -0,0 +1,38 @@
+# VFor with Index Key
+
+Checks if there are indexes being used as `key` in `v-for`.
+
+## ❓ Why it's good to follow this rule?
+
+- **Avoid Rendering Issues:** Using unique keys prevents issues with item reordering and re-rendering, ensuring stable and predictable updates.
+- **Improve Performance:** Unique keys help Vue.js efficiently update and manage list items, leading to better performance.
+- **Enhance Debugging:** Unique keys make it easier to track and resolve rendering issues, simplifying the debugging process.
+
+
+## 😱 Examples of code for which this rule will throw a warning
+
+::: warning
+The following code contains a `v-for` directive that uses the loop index as the `:key`, which is discouraged because it can cause rendering issues when the array changes.
+:::
+
+```vue
+<template>
+  <div v-for="(item, idx) in items" :key="idx">
+    {{ item.name }}
+  </div>
+</template>
+```
+
+## 🤩 How to fix it?
+
+::: tip
+To ensure stable rendering and improve performance, use a unique identifier from the items themselves as the `:key`. This could be an id or another unique property.
+:::
+
+```vue
+<template>
```

Comment:
I really like using the same examples in the docs and in the tests. Somehow it just feels right and consistent.

Perhaps it is not possible to all cases.

🤩

Reply:
Oh I see, the example in the docs is missing the <p> tag I think.

> Perhaps it is not possible to all cases

For example if this rule's tests have 3 where it should report error, you mean adding those three to the docs?

Or just taking one of the tests to be the docs example?

Reply:
One of them will work.

Reply:
Updated in latest commit https://github.com/rrd108/vue-mess-detector/commit/202a0c21e72cb82089088c7e6cbebd60678b4d89